File: backend/internal/models/enums.go
package models

type MediaType string

const (
	MediaTypeMovie         MediaType = "movie"
	MediaTypeTVSeries      MediaType = "tvSeries"
	MediaTypeAnimeMovie    MediaType = "animeMovie"
	MediaTypeAnimeSeries   MediaType = "animeSeries"
	MediaTypeCartoonMovie  MediaType = "cartoonMovie"
	MediaTypeCartoonSeries MediaType = "cartoonSeries"
	MediaTypeGame          MediaType = "game"
	MediaTypeManga         MediaType = "manga"
	MediaTypeBook          MediaType = "book"
	MediaTypeLightNovel    MediaType = "lightNovel"
)

type ReviewStatus string

const (
	ReviewStatusNeutral               ReviewStatus = "neutral"
	ReviewStatusPositive              ReviewStatus = "positive"
	ReviewStatusNegative              ReviewStatus = "negative"
	ReviewStatusSurprised             ReviewStatus = "surprised"
	ReviewStatusDisappointed          ReviewStatus = "disappointed"
	ReviewStatusExcited               ReviewStatus = "excited"
	ReviewStatusConfused              ReviewStatus = "confused"
	ReviewStatusLaugh                 ReviewStatus = "laugh"
	ReviewStatusBoring                ReviewStatus = "boring"
	ReviewStatusCry                   ReviewStatus = "cry"
	ReviewStatusCool                  ReviewStatus = "cool"
	ReviewStatusCold                   ReviewStatus = "cold"
	ReviewStatusScary                 ReviewStatus = "scary"
	ReviewStatusShushing              ReviewStatus = "shushing"
	ReviewStatusExplodingHead         ReviewStatus = "exploding_head"
	ReviewStatusFaceHoldingBackTears  ReviewStatus = "face_holding_back_tears"
	ReviewStatusKiss                  ReviewStatus = "kiss"
	ReviewStatusPleadingFace          ReviewStatus = "pleading_face"
	ReviewStatusSalutingFace          ReviewStatus = "saluting_face"
)

type AgeRating string

const (
	AgeRatingG    AgeRating = "g"
	AgeRatingPG   AgeRating = "pg"
	AgeRatingPG13 AgeRating = "pg13"
	AgeRatingR    AgeRating = "r"
	AgeRatingNC17 AgeRating = "nc17"
	AgeRatingNC21 AgeRating = "nc21"
	AgeRatingTVY  AgeRating = "tvY"
	AgeRatingTVY7 AgeRating = "tvY7"
	AgeRatingTVG  AgeRating = "tvG"
	AgeRatingTVPG AgeRating = "tvPg"
	AgeRatingTV14 AgeRating = "tv14"
	AgeRatingTVMA AgeRating = "tvMa"
)

type MediaStatus string

const (
	MediaStatusAnnounced    MediaStatus = "announced"    // Анонсировано
	MediaStatusInProduction MediaStatus = "in_production" // В производстве
	MediaStatusReleased     MediaStatus = "released"     // Вышло / выходит
	MediaStatusFinished     MediaStatus = "finished"     // Завершено
	MediaStatusCancelled    MediaStatus = "cancelled"    // Отменено
	MediaStatusPostponed    MediaStatus = "postponed"    // Отложено
)

type Profession string

const (
	ProfessionActor          Profession = "actor"
	ProfessionActress        Profession = "actress"
	ProfessionDirector       Profession = "director"
	ProfessionProducer       Profession = "producer"
	ProfessionWriter         Profession = "writer"
	ProfessionCinematographer Profession = "cinematographer"
	ProfessionComposer       Profession = "composer"
	ProfessionEditor         Profession = "editor"
	ProfessionAnimator       Profession = "animator"
	ProfessionVoiceActor     Profession = "voiceActor"
	ProfessionAuthor         Profession = "author"       // Книги, манга, ранобэ
	ProfessionIllustrator    Profession = "illustrator"  // Иллюстратор (книги, ранобэ)
	ProfessionArtist         Profession = "artist"       // Художник (манга, комиксы)
	ProfessionGameDesigner   Profession = "gameDesigner" // Геймдизайнер (игры)
	ProfessionLevelDesigner  Profession = "levelDesigner"
	ProfessionTranslator     Profession = "translator"   // Переводчик (книги, локализация)
	ProfessionLiteraryEditor Profession = "literaryEditor" // Литературный редактор (книги)
)

type RoleType string

const (
	RoleTypeMain       RoleType = "main"
	RoleTypeSupporting RoleType = "supporting"
	RoleTypeCameo      RoleType = "cameo"
	RoleTypeExtra      RoleType = "extra"
	RoleTypeNarrator   RoleType = "narrator"
	RoleTypeGuest      RoleType = "guest"
	RoleTypeBackground RoleType = "background"
)

type ListStatus string

const (
	ListStatusWatching   ListStatus = "watching"
	ListStatusPlanned    ListStatus = "planned"
	ListStatusCompleted  ListStatus = "completed"
	ListStatusOnHold     ListStatus = "onHold"
	ListStatusDropped    ListStatus = "dropped"
	ListStatusRewatching ListStatus = "rewatching"
)

type TitleReaction string

const (
	TitleReactionSurprised    TitleReaction = "surprised"    // 😲
	TitleReactionDisappointed TitleReaction = "disappointed" // 😞
	TitleReactionSad          TitleReaction = "sad"          // 😢
	TitleReactionJoyful       TitleReaction = "joyful"       // 😊
	TitleReactionInspiring    TitleReaction = "inspiring"    // ✨
	TitleReactionScary        TitleReaction = "scary"        // 😱
	TitleReactionFunny        TitleReaction = "funny"         // 😂
	TitleReactionAngry        TitleReaction = "angry"         // 😠
	TitleReactionLove         TitleReaction = "love"          // 😍
	TitleReactionNeutral      TitleReaction = "neutral"       // 😐
	TitleReactionBoring       TitleReaction = "boring"       // 😴
	TitleReactionExciting     TitleReaction = "exciting"     // 🤩
	TitleReactionTouching     TitleReaction = "touching"      // 🥲
	TitleReactionThoughtful   TitleReaction = "thoughtful"   // 🤔
)

func ValidTitleReaction(s string) bool {
	switch TitleReaction(s) {
	case TitleReactionSurprised, TitleReactionDisappointed, TitleReactionSad, TitleReactionJoyful,
		TitleReactionInspiring, TitleReactionScary, TitleReactionFunny, TitleReactionAngry,
		TitleReactionLove, TitleReactionNeutral, TitleReactionBoring, TitleReactionExciting,
		TitleReactionTouching, TitleReactionThoughtful:
		return true
	}
	return false
}

type AnimeSeason string

const (
	AnimeSeasonWinter AnimeSeason = "winter"
	AnimeSeasonSpring AnimeSeason = "spring"
	AnimeSeasonSummer AnimeSeason = "summer"
	AnimeSeasonAutumn AnimeSeason = "autumn"
)

type MediaRelationType string

const (
	MediaRelationPrequel            MediaRelationType = "prequel"
	MediaRelationSequel             MediaRelationType = "sequel"
	MediaRelationSpinOff            MediaRelationType = "spinOff"
	MediaRelationAlternativeVersion MediaRelationType = "alternativeVersion"
	MediaRelationSideStory          MediaRelationType = "sideStory"
	MediaRelationCrossover          MediaRelationType = "crossover"
	MediaRelationCompilation        MediaRelationType = "compilation"
	MediaRelationRemake             MediaRelationType = "remake"
	MediaRelationRemaster           MediaRelationType = "remaster"
	MediaRelationAdaptation         MediaRelationType = "adaptation"
)

File: frontend/src/constants/enums.ts
/**
 * Enums matching backend (internal/models/enums.go, internal/models/user.go).
 * Use for role types (cast), professions (person), and user roles.
 */

/** Роль пользователя (БД: user.role) */
export type UserRole = 'user' | 'admin' | 'moderator' | 'content_creator' | 'developer' | 'owner'

/** Все значения UserRole из БД */
export const USER_ROLES: readonly UserRole[] = [
  'user',
  'admin',
  'moderator',
  'content_creator',
  'developer',
  'owner',
] as const

export type RoleType = 'main' | 'supporting' | 'cameo' | 'extra' | 'narrator' | 'guest' | 'background'

/** Все значения RoleType из БД в порядке отображения */
export const ROLE_TYPES: readonly RoleType[] = [
  'main',
  'supporting',
  'cameo',
  'guest',
  'narrator',
  'extra',
  'background',
] as const

/** Профессия/роль в проекте (БД: profession у person, роль в касте) */
export type Profession =
  | 'actor'
  | 'actress'
  | 'director'
  | 'producer'
  | 'writer'
  | 'cinematographer'
  | 'composer'
  | 'editor'
  | 'animator'
  | 'voiceActor'
  | 'author'
  | 'illustrator'
  | 'artist'
  | 'gameDesigner'
  | 'levelDesigner'
  | 'translator'
  | 'literaryEditor'

/** Возрастной рейтинг (БД: age_rating enum) */
export type AgeRating = 'g' | 'pg' | 'pg13' | 'r' | 'nc17' | 'nc21' | 'tvY' | 'tvY7' | 'tvG' | 'tvPg' | 'tv14' | 'tvMa'

/** Все значения AgeRating из БД */
export const AGE_RATINGS: readonly AgeRating[] = [
  'g',
  'pg',
  'pg13',
  'r',
  'nc17',
  'nc21',
  'tvY',
  'tvY7',
  'tvG',
  'tvPg',
  'tv14',
  'tvMa',
] as const

/** Все значения Profession из БД */
export const PROFESSIONS: readonly Profession[] = [
  'actor',
  'actress',
  'director',
  'producer',
  'writer',
  'cinematographer',
  'composer',
  'editor',
  'animator',
  'voiceActor',
  'author',
  'illustrator',
  'artist',
  'gameDesigner',
  'levelDesigner',
  'translator',
  'literaryEditor',
] as const

/** Профессии для блока «Персонал» (без актёров и актёров озвучки — они в касте/дубляже) */
export const STAFF_PROFESSIONS: readonly Profession[] = [
  'director',
  'producer',
  'writer',
  'cinematographer',
  'composer',
  'editor',
  'animator',
  'author',
  'illustrator',
  'artist',
  'gameDesigner',
  'levelDesigner',
  'translator',
  'literaryEditor',
] as const

/**
 * Ключи для группировки работ персоны (person.works).
 * director, screenwriter, actor, dubbing — базовые; остальные из Profession.
 * writer отображается как screenwriter в UI (person.screenwriter).
 */
export type PersonWorkRoleKey =
  | 'director'
  | 'screenwriter'
  | 'actor'
  | 'actress'
  | 'dubbing'
  | 'producer'
  | 'writer'
  | 'cinematographer'
  | 'composer'
  | 'editor'
  | 'animator'
  | 'voiceActor'

/** Порядок отображения блоков по ролям на странице персоны */
export const PERSON_WORK_ROLE_KEYS: readonly PersonWorkRoleKey[] = [
  'director',
  'screenwriter',
  'actor',
  'actress',
  'dubbing',
  'producer',
  'writer',
  'cinematographer',
  'composer',
  'editor',
  'animator',
  'voiceActor',
] as const
